File: Medium/2196. Create Binary Tree From Descriptions.java
class Solution {
    public TreeNode createBinaryTree(int[][] descriptions) {
        final var map = new TreeNode[100001];
        final var set = new boolean[100001];
        TreeNode parent = null, child = null;
        for (var desc : descriptions) {
            parent = map[desc[0]] == null ? map[desc[0]] = new TreeNode(desc[0]) : map[desc[0]];
            child = map[desc[1]] == null ? map[desc[1]] = new TreeNode(desc[1]) : map[desc[1]];
            if (desc[2] > 0) {
                parent.left = child;
            } else {
                parent.right = child;
            }
            set[child.val] = true;
        }
        TreeNode root = null;
        for (var desc : descriptions) {
            if (!set[desc[0]]) {
                root = map[desc[0]];
                break;
            }
        }
        return root;
    }
}

File: Medium/2196. Create Binary Tree From Descriptions.py
# Definition for a binary tree node.
# class TreeNode:
#     def __init__(self, val=0, left=None, right=None):
#         self.val = val
#         self.left = left
#         self.right = right
class Solution:
    def createBinaryTree(self, descriptions: List[List[int]]) -> Optional[TreeNode]:
        node_dict = {}
        child_set = set()
        root = None
        for val, child, isLeft in descriptions:
            if val in node_dict:
                tree = node_dict[val]
            else:
                tree = TreeNode(val)
                node_dict[val] = tree
            if child in node_dict:
                child_tree = node_dict[child]
            else:
                child_tree = TreeNode(child)
                node_dict[child] = child_tree
            child_set.add(child_tree)
            if isLeft:
                tree.left = child_tree
            else:
                tree.right = child_tree
        for root_value, tree in node_dict.items():
            if tree not in child_set:
                return tree
